### PR TITLE
chore: use oasdiff instead of manual diff

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ CLOUDP-#

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/cli/internal/openapi/filter/path.go [hunk 1]
```diff
@@ -257,6 +259,13 @@ func filterVersionedContent(content map[string]*openapi3.MediaType, version *api
 	return nil, nil
 }
 
+func updateSingleMediaTypeExtension(m *openapi3.MediaType, version *apiversion.APIVersion) {
+	if m.Extensions == nil {
+		m.Extensions = make(map[string]interface{})
+	}
+	m.Extensions["x-xgen-version"] = version.String()
```
Comment (blva) [inline tools/cli/internal/openapi/filter/path.go:266]: fixing the version here in our files

File: tools/cli/test/e2e/cli/split_test.go [hunk 2]
```diff
@@ -53,3 +55,37 @@ func validateFiles(t *testing.T, version string) {
 	require.NoError(t, err)
 	ValidateVersionedSpec(t, NewValidAtlasSpecPath(t, version), path)
 }
+
+func ValidateVersionedSpec(t *testing.T, correctSpecPath, generatedSpecPath string) {
+	t.Helper()
+	correctSpec := newOpenAPISpec(t, correctSpecPath)
```
Comment (blva) [inline tools/cli/test/e2e/cli/split_test.go:61]: we can use this to validate all diffs, and incrementally add assertions

File: tools/cli/internal/openapi/filter/path.go [hunk 3]
```diff
@@ -79,7 +79,7 @@ func (f *PathFilter) apply(path *openapi3.PathItem, m *Metadata) error {
 			return err
 		}
 
-		removeDeprecatedReponses(op, config, opConfig)
+		updateReponses(op, config, opConfig)
```
Comment (andreaangiolillo) [inline tools/cli/internal/openapi/filter/path.go:82]:
```suggestion
		updateResponses(op, config, opConfig)
```

File: tools/cli/internal/openapi/filter/path.go [hunk 4]
```diff
@@ -257,6 +259,13 @@ func filterVersionedContent(content map[string]*openapi3.MediaType, version *api
 	return nil, nil
 }
 
+func updateSingleMediaTypeExtension(m *openapi3.MediaType, version *apiversion.APIVersion) {
```
Comment (andreaangiolillo) [inline tools/cli/internal/openapi/filter/path.go:262]: Could you add a small description of what the functions does?

Comment (blva) [reply tools/cli/internal/openapi/filter/path.go:262]: thats trash, deleted now!

Comment (blva) [reply tools/cli/internal/openapi/filter/path.go:262]: oh no, nvm thought it was a comparison one, will do!

File: tools/cli/test/e2e/cli/split_test.go [hunk 5]
```diff
@@ -49,7 +50,37 @@ func getOutputFolder(t *testing.T) string {
...
+	// compare w/ oasdiff
+	d, err := diff.Get(diff.NewConfig(), correctSpec, generatedSpec)
+	require.NoError(t, err)
+
+	message := "Generated spec is not equal to the correct spec for path: " + correctSpecPath + "\n\n" +
+		"git diff --no-index " + correctSpecPath + " " + generatedSpecPath + " > diff.diff"
+
+	require.Empty(t, d.ExtensionsDiff, message)
+	require.Empty(t, d.OpenAPIDiff, message)
+	// require.Empty(t, d.InfoDiff, message) TODO: add in next PR
```
Comment (blva) [inline tools/cli/test/e2e/cli/split_test.go:71]: next prs will be to uncoment one by one

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**